setup.py

```diff
@@ -67,6 +67,10 @@ def generate_proto(source):
         'numpy>=1.21',
         'opt_einsum',
         'scipy>=1.7',
+        # Required by xla_bridge.discover_pjrt_plugins for forwards compat with
+        # Python versions < 3.10. Can be dropped when 3.10 is the minimum
+        # required Python version.
+        'importlib_metadata>=4.6',
```

Comment:
Can you add this in extras or make the user install this explictly?

Reply:
This was my fear. I'll have to add more logic to make it optional. Is it bad to add an extra requirement?

Reply:
After offline discussion, we decided this dep is ok. We don't want to require users to manually install since it's necessary for plugin detection (at least via entry_points) and can cause jax import to fail.